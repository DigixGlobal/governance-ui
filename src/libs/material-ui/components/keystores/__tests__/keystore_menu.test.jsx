import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import KeystoreMenu from '../keystore_menu';

jest.mock('semantic-ui-react');

const keystoreTypes = [{
  id: 1,
  name: 'keystore test 1',
  icon: 'testIcon',
  color: 'blue',
  subtitle: 'test subtitle',
}, {
  id: 2,
  name: 'keystore test 2',
  icon: 'testIcon2',
  color: 'black',
  subtitle: 'test subtitle 2',
}];

describe('<KeystoreMenu />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <KeystoreMenu
        formChange={jest.fn}
        keystoreTypes={keystoreTypes}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
