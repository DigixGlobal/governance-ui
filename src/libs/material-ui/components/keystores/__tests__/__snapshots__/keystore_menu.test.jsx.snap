// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KeystoreMenu /> renders correctly 1`] = `
<Menu
  icon="labeled"
  stackable={true}
  widths={2}
>
  <MenuItem
    as="a"
    key="1"
    name={1}
    onClick={[Function]}
  >
    <Header
      as="h4"
      icon={true}
      style={
        Object {
          "height": "auto",
        }
      }
    >
      <Icon
        as="i"
        color="blue"
        name="testIcon"
      />
      keystore test 1
      <HeaderSubheader>
        test subtitle
      </HeaderSubheader>
    </Header>
  </MenuItem>
  <MenuItem
    as="a"
    key="2"
    name={2}
    onClick={[Function]}
  >
    <Header
      as="h4"
      icon={true}
      style={
        Object {
          "height": "auto",
        }
      }
    >
      <Icon
        as="i"
        color="black"
        name="testIcon2"
      />
      keystore test 2
      <HeaderSubheader>
        test subtitle 2
      </HeaderSubheader>
    </Header>
  </MenuItem>
</Menu>
`;
